routes/common/load_balancer_helper.rb

```diff
@@ -41,6 +41,16 @@ def post(name: nil)
       end
     end
     Authorization.authorize(@user.id, "PrivateSubnet:view", ps.id)
+    prefix, dns_zone_id = if (custom_hostname = request_body_params["custom_hostname"])
+      dns_zone_name = custom_hostname.split(".")[1..].join(".")
+      prefix = custom_hostname.split(".")[0]
+      dns_zone = DnsZone.where(name: dns_zone_name).first
```

Comment:
This allows user to modify arbitrary dns zone that they don't necessarily own.

Load balancer's custom_hostname_dns_zone logic is quite scary. (Sorry for just catching that, I missed the original PR). I'm not comfortable how it is possible to pass an arbitrary DnsZone id to LoadBalancerNexus and it takes control of the DnsZone. DnsZones are like the keys of the whole house. It is possible to do massive damage using them.

One proper way could be load balancer to create its own DnsZone and manage only that.

Reply:
Yeah, I was staring at that and asking furkan a few questions about the scoping of this. seems hairy as-is

Reply:
Please recheck the commit message.